crates/maybenot-ffi/maybenot.h

```diff
@@ -0,0 +1,181 @@
+/* Generated with cbindgen:0.26.0 */
+
+/* Warning, this file is autogenerated by cbindgen. Don't modify this manually. */
```

Comment:
If the header is auto generated I don't think it should be checked in to git. It's either or, not both IMO. Automatically generated content does not belong in git, unless re-generating it is a manual step very seldom done.

Reply:
I disagree, my reason for including it is transparency. Having the `.h`-file checked in makes it very easy to inspect the FFI without having to clone and build the repo. Do you see any concrete drawbacks to having it checked in?

Reply:
> patch release change to cbindgen

Fair point, we can fix that by pinning the cbindgen version in Cargo.toml.
```toml
cbindgen = "=0.26.0"
```

> building a project should not change source code

Barring any cbindgen patch releases, this will only happen if you've already changed some other part of the source code. I think that's fine, just like how Cargo.lock must be updated, if you've changed Cargo.toml.

Admittedly, if someone commits and pushes changes to the Rust API *without* running cargo check, cargo build, or even rust-analyzer, then the .h-file will become stale. That would be annoying, but I think it would be such an unlikely occurence that the risk is worth it.

Reply:
Exact version pinning has other downsides. I really try to avoid it. If any other crate in the *entire dependency tree* pins another version that is semver compatible with this version, for example `0.26.1`, then cargo will not be able to compile anything at all, because it can't select what version of `cbindgen` to pick.

Reply:
Ah... that's not good

Reply:
But I don't see any downside with commiting the entire workspace `Cargo.lock`. The official guidelines on this changed rather recently. It is now fully OK for libraries to do so if it suits their needs. It also makes it easier to reproduce certain issues. So I think libraries most often should commit `Cargo.lock`.

Reply:
Cool, I went ahead and did that